Spatial_searching/include/CGAL/Kd_tree.h

```diff
@@ -128,50 +149,69 @@ class Kd_tree {
   Node_handle
   create_leaf_node(Point_container& c)
   {
-    Leaf_node node(true , static_cast<unsigned int>(c.size()));
+    Leaf_node node(static_cast<unsigned int>(c.size()));
     std::ptrdiff_t tmp = c.begin() - data.begin();
     node.data = pts.begin() + tmp;
 
-    leaf_nodes.push_back(node);
-    Leaf_node_handle nh = &leaf_nodes.back();
-
-
-    return nh;
+#ifdef CGAL_TBB_STRUCTURE_IN_KD_TREE
+    return &*(leaf_nodes.push_back(node));
+#else
+    leaf_nodes.emplace_back (node);
+    return &(leaf_nodes.back());
+#endif
   }
 
-
   // The internal node
-
-  Node_handle
-  create_internal_node(Point_container& c, const Tag_true&)
-  {
-    return create_internal_node_use_extension(c);
-  }
-
-  Node_handle
-  create_internal_node(Point_container& c, const Tag_false&)
+  Node_handle new_internal_node()
   {
-    return create_internal_node(c);
+#ifdef CGAL_TBB_STRUCTURE_IN_KD_TREE
+    return &*(internal_nodes.push_back(Internal_node()));
+#else
+    internal_nodes.emplace_back (Internal_node());
```

Comment:
I do not understand that use of `emplace_back`: why use the copy-constructor from a default-constructed `Internal_node`? I suggest:
```suggestion
    internal_nodes.emplace_back ();
```
That passes the empty set of argument to the constructor of `Internal_node`, constructing a default-constructed `Internal_node` inside the container.